net/interfaces/interfaces.go

```diff
@@ -81,13 +81,17 @@ func isProblematicInterface(nif *net.Interface) bool {
 }
 
 // LocalAddresses returns the machine's IP addresses, separated by
-// whether they're loopback addresses.
+// whether they're loopback addresses. Separately for IPv4 and IPv6, if
+// there are no regular addresses it will return IPv4 linklocal or
+// IPv6 unique linklocal addresses because we know of environments
```

Comment:
ULA is Unique Local Address .... I don't think it's a _link_ local.

Reply:
Done.